lang/python/python-cryptography/patches/0004-make-rust-completely-optional-during-build.patch

```diff
@@ -0,0 +1,109 @@
+--- a/setup.py
++++ b/setup.py
+@@ -10,8 +10,11 @@ import sys
+ 
+ from setuptools import find_packages, setup
+ 
++dont_build_rust = os.environ.get("CRYPTOGRAPHY_DONT_BUILD_RUST")
++
+ try:
+-    from setuptools_rust import RustExtension
++    if (not dont_build_rust):
++        from setuptools_rust import RustExtension
+ except ImportError:
+     print(
+         """
+@@ -41,13 +44,48 @@ with open(os.path.join(src_dir, "cryptog
+ 
+ # `install_requirements` and `setup_requirements` must be kept in sync with
+ # `pyproject.toml`
+-setuptools_rust = "setuptools-rust>=0.11.4"
+ install_requirements = ["cffi>=1.12"]
+-setup_requirements = install_requirements + [setuptools_rust]
++setup_requirements = install_requirements
++
++extra_requirements = {
```

Comment:
This patch looks like it is correct to make Rust completely optional, but I think it could be a lot smaller if we instead just disable Rust completely (skip the `RustExtension` import test, remove `setuptools_rust` from `setup_requirements`). (I don't think upstream will ever accept this patch anyway.)

In particular, `extras_require` can be left alone as these dependencies are not installed by default. (To install these pip would need to be invoked like `pip install cryptography[test]` or `pip install cryptography[sdist]`.)

Reply:
yeah; you're right;
this patch was initially made to be sent upstream, then i saw that they removed the `CRYPTOGRAPHY_DONT_BUILD_RUST`  stuff in master [after i made this];